.gitlab/helper_functions.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+
+CYAN="\e[0;36m"
+CLEAR="\e[0m"
+SECTION_START="\e[0Ksection_start:the_time:section_id\r\e[0K${CYAN}section_header${CLEAR}"
+SECTION_END="\e[0Ksection_end:the_time:section_id\r\e[0K"
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+BLUE='\033[0;34m'
+NC='\033[0m'
+
+section_start() {
+    local section_header section_id start
+    start="$SECTION_START"
+    if [[ "$#" -eq 1 ]]; then
+        section_header="$1"
+        section_id="$(echo "$1" | tr -c '[:alnum:]\n\r' '_')"
+    elif [[ "$#" -eq 2 ]]; then
+        if [[ "$2" =~ -{0,2}collapsed ]]; then
+            start="${start/section_id/section_id[collapsed=true]}"
+            section_header="$1"
+            section_id="$(echo "$1" | tr -c '[:alnum:]\n\r' '_')"
+        else
+            section_header="$2"
+            section_id="$1"
+        fi
+    elif [[ "$#" -eq 3 && "$3" =~ /^-{0,2}collapsed$/ ]]; then
+        start="${start/section_id/section_id[collapsed=true]}"
+        section_header="$2"
+        section_id="$1"
+    else
+        echo "section_start should be called with 1-3 args but it was called with $#"
+        echo "acceptable usages:"
+        echo "    1. section_start \"<section-start-id>\" \"<section-header>\""
+        echo "    2. section_start \"<section-header>\""
+        echo "    3. section_start \"<section-start-id>\" \"<section-header>\" --collapse"
+        echo "    4. section_start \"<section-header>\" --collapse"
```

Comment:
Suggest to group by functionality
```suggestion
        echo "    2. section_start \"<section-header>\""
        echo "    4. section_start \"<section-header>\" --collapse"
        echo "    1. section_start \"<section-start-id>\" \"<section-header>\""
        echo "    3. section_start \"<section-start-id>\" \"<section-header>\" --collapse"
```